environments/jvm/Dockerfile

```diff
@@ -1,7 +1,13 @@
 FROM maven:3.5-jdk-8 as BUILD
-COPY src /usr/src/myapp/src
-COPY pom.xml /usr/src/myapp
-RUN mvn -f /usr/src/myapp/pom.xml clean package
+WORKDIR /usr/src/myapp/
+
+# To reuse the build cache, here we split maven dependency
+# download and package into two RUN commands to avoid cache invalidation.
+COPY pom.xml .
+RUN mvn dependency:go-offline
```

Comment:
I am sorry I don't understand how this will avoid cache invalidation. AFAIK - go-offline is noly downloading the libraries offline